proto/ibc/core/client/v2/counterparty.proto

```diff
@@ -0,0 +1,43 @@
+syntax = "proto3";
+
+package ibc.core.client.v2;
+
+option go_package = "github.com/cosmos/ibc-go/v9/modules/core/02-client/types";
+
+import "cosmos/msg/v1/msg.proto";
+import "gogoproto/gogo.proto";
+
+// CounterpartyInfo defines the key that the counterparty will use to message our client
+message CounterpartyInfo {
+  // merkle prefix key is the prefix that ics provable keys are stored under
+  repeated bytes merkle_prefix = 1;
```

Comment:
I decided to just make this `repeated bytes` rather than importing ics23.commitment MerklePath

I think the inclusion of ics23 into the required external structs has caused more confusion than anything else. It is also an unnecessary import here

I convert to ics23 commitment MerklePath internally

Reply:
I think it is ok to have this be `ics23.commitment.MerklePath` but use `repeated bytes` in `MsgRegisterCounterparty` since `CounterpartyInfo` is used internally anyway?

But this is also good :)